R/data-combined.R

```diff
@@ -0,0 +1,625 @@
+#' @importFrom ospsuite.utils validateIsLogical
+#' @importFrom ospsuite.utils validateIsNumeric
+#' @importFrom ospsuite.utils validateIsOfLength
+#' @importFrom ospsuite.utils validateIsOfType
+#' @importFrom ospsuite.utils validateIsSameLength
+#' @importFrom ospsuite.utils validateIsString
+#' @importFrom ospsuite.utils validatePathIsAbsolute
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## constructor method ---------------
+
+    #' @description
+    #' Initialize a new instance of the class.
+    #' The constructor is empty because we prefer that the users add
+    #' `SimulationResults` and `DataSet` objects separately and individually.
+    #' @return A new instance of `DataCombined` object.
+
+    # empty constructor
+    initialize = function() {
+    },
+
+    ## setter methods ---------------
+
+    #' @description
+    #' Add simulated data.
+    #' @return `DataCombined` object containing simulated data.
+
+    addSimulationResults = function(simulationResults,
+                                    groups = NULL,
+                                    names = NULL,
+                                    quantitiesOrPaths = NULL,
+                                    population = NULL,
+                                    individualIds = NULL) {
+      # fail fast; walk is needed because there is no output here
+      purrr::walk(.x = groups, .f = ~ validateIsString(.x, nullAllowed = TRUE))
+
+      # list of `SimulationResults` instances is not allowed
+      if (is.list(simulationResults)) {
+        stop(
+          "Only a single instance, and not a list, of `SimulationResults` objects is expected.",
+          call. = FALSE
+        )
+      }
+
+      # validate the object type
+      validateIsOfType(simulationResults, SimulationResults)
+
+      # make sure that length of groups and names is the same
+      # additionally since these lists are going to have strings as elements,
+      # convert NULLs to NAs
+      groups <- .validateListArgs(groups, simulationResults$allQuantityPaths)
+      names <- .validateListArgs(names, quantitiesOrPaths %||% simulationResults$allQuantityPaths)
+
+      # save the original object as it is; useful for `$toDataFrame()` method
+      # styler: off
+      private$.simulationResults              <- simulationResults
```

Comment:
It does not make sense like this, because we can add multiple simulation results (one after the other). Actually, once the results have been added, we do not need them any more, so no need to store in the object?